pipeline_dp/dp_computations.py

```diff
@@ -70,6 +145,40 @@ def _add_random_noise(value: float, eps: float, delta: float,
     raise ValueError("Noise kind must be either Laplace or Gaussian.")
 
 
+def equally_split_budget(eps: float, delta: float, no_mechanisms: int):
+    """Equally splits the budget (eps, delta) between a given number of mechanisms.
+
+    Args:
+        eps, delta: The available budget.
+        no_mechanisms: The number of mechanisms between which we split the budget.
+
+    Raises:
+        ValueError: The number of mechanisms must be a positive integer.
+
+    Returns:
+        An array with the split budgets.
+    """
+    if no_mechanisms <= 0:
+        raise ValueError(
+            "The number of mechanisms must be a positive integer.")
+
+    # These variables are used to keep track of the budget used instead
+    # of estimating it. In this way, we can make use of the whole budget.
```

Comment:
Nit: remove "instead of estimating it." (it's not clear what estimating mean).
Please change it: In this way, we can improve accuracy of floating-point operations.